src/platform/unittest/TlsTest.cpp

```diff
@@ -55,14 +58,23 @@ struct TlsTest : public ::testing::TestWithParam<bool>
 
     static void SetUpTestSuite()
     {
-        SelfSignedCertParams = CxPlatPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CERT_USER);
+        SelfSignedCertParams = (QUIC_CREDENTIAL_CONFIG*)CxPlatPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CERT_USER, FALSE);
         ASSERT_NE(nullptr, SelfSignedCertParams);
+#ifndef QUIC_DISABLE_CLIENT_CERT_TESTS
+        ClientCertParams = (QUIC_CREDENTIAL_CONFIG*)CxPlatPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CERT_USER, TRUE);
+        ASSERT_NE(nullptr, ClientCertParams);
+        ClientCertParams->Flags |= QUIC_CREDENTIAL_FLAG_NO_CERTIFICATE_VALIDATION;
+#endif
     }
 
     static void TearDownTestSuite()
     {
         CxPlatPlatFreeSelfSignedCert(SelfSignedCertParams);
         SelfSignedCertParams = nullptr;
+#ifndef QUIC_DISABLE_CLIENT_CERT_TESTS
```

Comment:
Don't actually need these `ifndef` checks here.

Reply:
`CxPlatPlatFreeSelfSignedCert` dereferences the parameter to get the `PCCERT_CONTEXT`, which leads to a null deref crash. These are required to avoid that crash.